shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -0,0 +1,236 @@
+    } catch (AlluxioException e) {
+      throw new IOException(e.getMessage());
+    }
+
+    if (srcStatus.isFolder()) {
+      if (dstStatus != null && !dstStatus.isFolder()) {
+        throw new IOException(
+            ExceptionMessage.DESTINATION_FILE_CANNOT_EXIST_WITH_WILDCARD_SOURCE.getMessage());
+      }
+
+      if (dstStatus == null) {
+        try {
+          mFileSystem.createDirectory(dstPath);
+          System.out.println("Created directory: " + dstPath.getPath());
+        } catch (AlluxioException e) {
+          throw new IOException("Fail to create directory: " + dstPath.getPath());
+        }
+      }
+
+      List<URIStatus> statuses;
+      try {
+        statuses = mFileSystem.listStatus(srcPath);
+      } catch (AlluxioException e) {
+        throw new IOException(e.getMessage());
+      }
+
+      List<String> errorMessages = new ArrayList<>();
+      for (URIStatus status : statuses) {
+        try {
+          copy(new AlluxioURI(srcPath.getScheme(), srcPath.getAuthority(), status.getPath()),
+              new AlluxioURI(dstPath.getScheme(), dstPath.getAuthority(),
+                  PathUtils.concatPath(dstPath.getPath(), status.getName())));
+        } catch (IOException e) {
+          errorMessages.add(e.getMessage());
+        }
+      }
+
+      if (errorMessages.size() != 0) {
+        throw new IOException(Joiner.on('\n').join(errorMessages));
+      }
+    } else {
+      copyFile(srcPath, dstPath);
+    }
+  }
+
+  /**
+   * Copies a file in the Alluxio filesystem.
+   *
+   * @param srcPath the source {@link AlluxioURI} (has to be a file)
+   * @param dstPath the destination path in the Alluxio filesystem
+   * @throws IOException
+   */
+  private void copyFile(AlluxioURI srcPath, AlluxioURI dstPath) throws IOException {
+    URIStatus srcStatus;
+    try {
+      srcStatus = mFileSystem.getStatus(srcPath);
+    } catch (AlluxioException e) {
+      throw new IOException(e.getMessage());
+    }
+
+    try (Closer closer = Closer.create()) {
+      OpenFileOptions openFileOptions = OpenFileOptions.defaults().setReadType(ReadType.NO_CACHE);
+      FileInStream is = closer.register(mFileSystem.openFile(srcPath, openFileOptions));
+      CreateFileOptions createFileOptions = CreateFileOptions.defaults();
+      if (srcStatus.isPersisted()) {
+        if (srcStatus.getInMemoryPercentage() == 0) {
+          createFileOptions.setWriteType(WriteType.THROUGH);
+        } else {
+          createFileOptions.setWriteType(WriteType.CACHE_THROUGH);
+        }
+      } else {
+        createFileOptions.setWriteType(WriteType.MUST_CACHE);
+      }
+      FileOutStream os = closer.register(mFileSystem.createFile(dstPath, createFileOptions));
+      IOUtils.copy(is, os);
+      System.out.println("Copied " + srcPath + " to " + dstPath.getPath());
+    } catch (AlluxioException e) {
+      throw new IOException(e.getMessage());
+    }
+  }
+
+  @Override
+  public String getUsage() {
```

Comment:
the current style follows the behavior of `CopyFromLocal` / `CopyToLocal` ... are you suggesting I add -R options there as well?